### PR TITLE
BN #24 Chores: set continous integration and test 

### PR DESCRIPTION
**Pull request by Niyobuzima**
# PR Description
Please include a brief description of the pull request you want to raise.
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] Configured Testing environment with either Mocha, Chai and Sinon, or Jest and Supertest to ensure that all functionalities are tested before deployment.
- [x] GitHub Actions is configured as a continuous integration service used to build and test the app.
- [x] Configured a deployment pipeline on Cyclic or Render, or any other host
# Description of tasks that were expected to be completed and that are not yet completed
- [ ] 
- [ ] 
- [ ]
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

**Review by uwdjany — DISMISSED** [review]

File: .github/workflows/gitActions.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Node.js CI
```
Comment (mukunzidd) [inline .github/workflows/gitActions.yml]: Name this just CI or Build

File: .gitignore
Comment (mukunzidd) [inline .gitignore:1]: Add .DS_Store in `gitignore`

File: .github/workflows/gitActions.yml
Comment (mukunzidd) [inline .github/workflows/gitActions.yml:1]: rename this file to CI so that it makes sense. The name should be more specific.

File: package.json [hunk 2]
```diff
@@ -3,37 +3,49 @@
   "version": "1.0.0",
   "description": "",
   "main": "index.js",
+  "type": "module",
   "scripts": {
-    "migrate": "npx sequelize-cli db:migrate",
-    "down": "npx sequelize-cli db:migrate:undo",
+    "start": "nodemon --exec babel-node src/index",
```
Comment (mukunzidd) [inline package.json]: - babel-node should not run in production
- run a normal node process

File: package.json [hunk 3]
```diff
@@ -3,37 +3,49 @@
...
     "i18next": "^22.4.13",
     "i18next-fs-backend": "^2.1.1",
     "i18next-http-middleware": "^3.3.0",
     "pg": "^8.10.0",
     "pg-hstore": "^2.3.4",
     "sequelize": "^6.30.0",
     "swagger-jsdoc": "^6.2.8",
-    "swagger-ui-express": "^4.6.2"
+    "swagger-ui-express": "^4.6.2",
+    "yaml": "^2.2.1"
```
Comment (mukunzidd) [inline package.json]: What exactly are you using this for?

File: package.json [hunk 4]
```diff
@@ -3,37 +3,49 @@
...
   "keywords": [],
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "@babel/cli": "^7.21.0",
+    "@babel/core": "^7.21.3",
+    "@babel/node": "^7.20.7",
+    "@babel/plugin-transform-runtime": "^7.21.0",
+    "@babel/preset-env": "^7.20.2",
+    "@babel/runtime": "^7.21.0",
```
Comment (mukunzidd) [inline package.json:29]: Why duplicate @babel packages both in dev and normal dependencies. Please fix this. [Ref](https://www.freecodecamp.org/news/setup-babel-in-nodejs/)

File: package.json [hunk 5]
```diff
@@ -3,37 +3,49 @@
...
-    "swagger-ui-express": "^4.6.2"
+    "swagger-ui-express": "^4.6.2",
+    "yaml": "^2.2.1"
   },
   "devDependencies": {
+    "eslint": "^8.37.0",
     "@babel/cli": "^7.21.0",
     "@babel/core": "^7.21.3",
     "@babel/node": "^7.20.7",
     "@babel/plugin-transform-runtime": "^7.21.0",
```
Comment (mukunzidd) [inline package.json]: Why exactly do we need this in this repo?

**Review by mukunzidd — CHANGES_REQUESTED** [review]
- Please @Niyobuzima do not mindlessly paste your PR description. Write relevant content and add a link to the relevant user user story on Trello. 
- Update the PR title. Use the format posted on the channel today. 
- Avoid grammatical error in either the title and the description. Is it set or setup?
- Reduce all the 17 commits into one comprehensive commit. Use an interactive git rebase and squash the commits. And write the new commit message according to the accepted format. Be careful lest you lose your work.

File: package.json (hunk 2 above)
Comment (Niyobuzima) [reply package.json]: so this feature is already on the develop so while pulling it was also included so i can't change

File: package.json (hunk 5 above)
Comment (Illustre13) [reply package.json]: think we should have this for a common formatting style

**Review by Illustre13 — DISMISSED** [review]
Well Done!!

**Review by Niyobuzima — COMMENTED** [review]
all changes have been made

**Review by Niyobuzima — COMMENTED** [review]
all changes have been made

File: .codeclimate.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  method-lines:
+    enabled: false
+  nested-control-flow:
+    enabled: false
+  return-statements:
+    enabled: false
+  similar-code:
+    enabled: false
+  identical-code:
+    enabled: false
```
Comment (dushimeemma) [inline .codeclimate.yml:22]: you can add an empty line at the end of all the files

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,2 @@
+{
+}
```
Comment (dushimeemma) [inline .vscode/settings.json:2]: add empty line as well

**Review by dushimeemma — DISMISSED** [review]

**Review by dushimeemma — COMMENTED** [review]
swatch the 91 commits into one commit and we are good to go

File: .github/workflows/CI.yml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run linter
+        run: npm run lint
+
+      - name: Run tests
+        run: npm test
```
Comment (Illustre13) [inline .github/workflows/CI.yml:55]: Add an empty line to the end of file

File: .sequelizerc [hunk 9]
```diff
@@ -2,7 +2,7 @@ require("@babel/register");
 const path = require("path");
 
 module.exports = {
-  "config": path.resolve("./src/database/config/config.json"),
+  "config": path.resolve("./src/database/config/config.js"),
   "models-path": path.resolve("./src/database/models"),
   "seeders-path": path.resolve("./src/database/seeders"),
   "migrations-path": path.resolve("./src/database/migrations"),
```
Comment (Illustre13) [inline .sequelizerc:8]: Add empty line at the of file

**Review by Illustre13 — CHANGES_REQUESTED** [review]
Add empty line at the end of file

**Pull request closed without merging**